### PR TITLE
feat(dashboard): Dashboard template story example

### PR DESCRIPTION
**Pull request by matthill8286**
## TL;DR

-  Created an example of a new Dashboard Template using the component library
- Added title to steps as a conditional and optional type 
- Updated loading states for the LazyLoadImage
- Added Loading states for the playlist title and description
- Extended the type `Asset` in `@/types/asset`

## Details

- 2 stories, a default generic dashboard with all sliders using 4 per screen, a secondary story where the sliders are using various sizes per screen,, all fully responsive.

## Screenshots

Some loading state, some are various sizes and default 4 / per dashboard layout.

Screenshot 2020-11-19 at 16 11 48

Screenshot 2020-11-19 at 16 15 06

####  Next Image

Screenshot 2020-11-19 at 16 14 50

####  Next Image

Screenshot 2020-11-19 at 16 14 43

####  Next Image

Screenshot 2020-11-19 at 16 14 14

####  Next Image

Screenshot 2020-11-19 at 16 14 02

####  Next Image

Screenshot 2020-11-19 at 16 13 05

Checklist / Reminders:

-   [X] Documentation / Story is created / adapted
-   [ ] Unit tests
-   [ ] CSS changes are BEM-compliant

### CODE REVIEW

**Review by eTheMan80 — APPROVED** [review]

**Review by tricaalexandru — APPROVED** [review]
@matthill8286 not seeing the playlist section component. 
I'm guessing the playlist section and playlist components are coming later, right?

**Comment by matthill8286** [comment]
> @matthill8286 not seeing the playlist section component.
> I'm guessing the playlist section and playlist components are coming later, right?

@tricaalexandru I'm not sure I'm following?? What's the playlist section component, is this the slider with title & copy underneath?? This is an example of the grid layout, it has 5 playlist slider components with arrows, and asset tiles that are lazy loaded, you can see here in their loading state https://user-images.githubusercontent.com/48310431/99678466-824b7e00-2a83-11eb-9bc0-83b608c7d808.png

Row 1 4 tiles for desktop
Row 2  3 tiles for desktop
Row 3 4 tiles
Row 4 3 tiles

And here https://user-images.githubusercontent.com/48310431/99678454-8081ba80-2a83-11eb-8638-7465a2fc0135.png  they are populated with title and description and the slider component of Row 2.

This is definitely something we can demo as an indication of how easy it is to throw something together.

**Comment by tricaalexandru** [comment]
@matthill8286 cool. The playlist section is the bit that encapsulates playlist of a certain type. 
So:
Explore playlists
Featured playlists
NexRex playlists etc. 

Those will only have title, description through tooltip and inside it the lazy loading happens. 
Does that make sense? We can look over the. existing UI so I can show you exactly. 

But it's ok, that can come in during our dashboard project and it will basically only encapsulate the playlists. No business logic within it.

**Comment by matthill8286** [comment]
Well then no, that isn't going to be built in the component library,
that'll be built when the dashboard project kicks off in the main magpie
app and plus there is no ticket for it now. 🙂.

We aren't going to lazy load the assets, we will be lazy loading the
playlists and main tiles in view ie default 4, as the playlist calls will
run asynchronously.

When we get to that project it'll become clearer, but on a good note we
have the dashboard container grid system done in storybook. I'll create the
onboarding one as it'll take me like 2hrs if that.

**Review by philjonasfiltered — APPROVED** [review]
